mlflow/azureml/__init__.py

```diff
@@ -311,6 +469,32 @@ def _get_mlflow_azure_resource_name():
             max_length=(azureml_max_resource_length - len(resource_prefix)))
     return resource_prefix + unique_id
 
+def _get_mlflow_azure_name_from_run(run_id):
+    """
+    :return: A unique name for an Azure resource indicating that the resource was created by
+             MLflow
+    """
+    azureml_max_resource_length = 32
+    resource_prefix = "mlflow-run-"
+    suffix = run_id[:azureml_max_resource_length-len(resource_prefix)]
+    return resource_prefix + suffix
+  
+
+def _create_mlflow_wheel(mlflow_dir, out_dir):
+    """
+    Create the wheel of MLFlow by using setup.py bdist_wheel in the outdir.
+
+    :param model_path: The absolute path to the mlflow.
+    :param out_dir: The absolute path to the outdir.
+    :return: The absolute path to the wheel.
+    """
+    out_path = Path(out_dir).resolve()
```

Comment:
I'm pretty sure that `out_dir` has to exist before `Path(out_dir).resolve()` will work (at least on Python 2); we're currently passing `out_dir = TempDir().path("dist")`, but the `dist` subdir does not yet exist.

The following small example fails in Python 2:
```
 from mlflow.utils.file_utils import TempDir
 with TempDir() as t:
     p = t.path("dist")
     from pathlib import Path
     Path(p).resolve()

OSError: [Errno 2] No such file or directory: '/private/var/folders/rq/g5vkbkv93r12yb1t374lfj0r0000gn/T/tmpNHXXk7/dist'
```

Reply:
Added making the directory.

semirelateldly, the  whole package doesn't support python2 per 289, although I guess I'm not entirely clear what that message means.